buttons: Improvements for action buttons and icon buttons.

sayamsamal opened the pull request:
This PR follows up on the discussions on [CZO](https://chat.zulip.org/#narrow/channel/101-design/topic/UI.20redesign.3A.20buttons.20.2332458/near/2013220) and #32791, and adds design fixes to the action buttons, and uses `clip-path` with transitions to improve the animation of the button clicks.

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
| Before | After |
|--------|--------|
| ![Screenshot 2025-01-04 at 9 55 24 AM](https://github.com/user-attachments/assets/536e412a-edee-4f53-9aaa-82175c68b2cd) | ![Screenshot 2025-01-04 at 9 50 36 AM](https://github.com/user-attachments/assets/d0950d9c-3c34-4a8c-ba9d-97a740010ebe) | 
| ![Screenshot 2025-01-04 at 9 56 04 AM](https://github.com/user-attachments/assets/020e4ffe-5141-486e-81f4-fc397fbf3af2) | ![Screenshot 2025-01-04 at 9 57 16 AM](https://github.com/user-attachments/assets/0dbb2423-a70b-4869-8afd-83d112afe1b0) | 

### State animations (Before)

https://github.com/user-attachments/assets/d355b7d5-1862-4071-8ad2-541f6777e4de

### State animations (After)

https://github.com/user-attachments/assets/007bdaad-9068-483b-9eff-06cce740d9f2

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
@karlstolley any feedback on this one?

terpimost commented:
Hey, thank you for preparing PR. Here are my notes:

Icon buttons. They are supposed to have icon in the center horizontally and vertically. So If you set width or height for this button the icon should be in the middle. It isn't the case now. You kind of having it right for regular buttons with labels.

I don't think --color-inner-shadow-action-button is switching properly in dark mode.

Once this is done I suggest @karlstolley to review it.

I would suggest to move forward with it assuming that at any point we might want to introduce a change or improvement to those components and change should be agreed on and reflected in /devtools/buttons

Example of future changes:
- loading indicator (I need to design this)
- specialized focus state (for now we are using default browser rect)
- disabled state as 50% opacity and no pointer events (but this is messing up tooltips, so how do we do that?)

sayamsamal commented:
## Update
- Updated the color values for action buttons from hex -> hsl.
- Fixed action button inner shadow in dark mode, and updated the same in light mode.

| Before | After |
|--------|--------|
| <img width="246" alt="Screenshot 2025-01-10 at 8 58 21 PM" src="https://github.com/user-attachments/assets/4aa74a2b-17cb-401d-b3ee-6af8e3288de4" /> | <img width="246" alt="Screenshot 2025-01-10 at 8 56 50 PM" src="https://github.com/user-attachments/assets/a7b57dc4-922a-4245-a5b1-1c683e821f70" /> |
| <img width="246" alt="Screenshot 2025-01-10 at 8 58 39 PM" src="https://github.com/user-attachments/assets/5e0ac1bc-5c05-4cc7-a953-1174f7153de6" /> | <img width="246" alt="Screenshot 2025-01-10 at 8 56 03 PM" src="https://github.com/user-attachments/assets/90a3c6c4-4f5e-4991-81ec-4442a8b8865b" /> | 

- Fixed alignment of icon inside the icon-button button styles.

| Before | After |
|--------|--------|
| <img width="114" alt="Screenshot 2025-01-10 at 9 00 32 PM" src="https://github.com/user-attachments/assets/f30203f5-3c62-4790-99a8-9b78d188a538" /> | <img width="114" alt="Screenshot 2025-01-10 at 9 01 30 PM" src="https://github.com/user-attachments/assets/8b8e4f8b-73ca-42ea-9ce7-3b529dd4dd7a" /> |

sayamsamal commented:
@karlstolley I have addressed the feedback from @terpimost on the PR. Could you please review this PR now?

karlstolley commented:
@sayamsamal I keep returning to review this, but it's in need of a rebase.

sayamsamal commented:
@karlstolley the commits in this PR have been merged via #32852, which was rebased on top of this. Please let me know if you find some discrepancies, on these commits in #32852, and I'll be sure to follow through with a follow up.

karlstolley commented:
Will do. Should this PR then be closed?
